### PR TITLE
Call popup resize handlers if the window innerHeight is changed (T834502)

### PR DESCRIPTION
**Pull request by alexander-kotov-dx**

### CODE REVIEW

File: js/core/utils/resize_callbacks.js [hunk 1]
```diff
@@ -18,13 +18,15 @@ var resizeCallbacks = (function() {
         var documentElement = domAdapter.getDocumentElement();
         return {
             width: documentElement.clientWidth,
-            height: documentElement.clientHeight
+            height: documentElement.clientHeight,
+            widthIOS: windowUtils.getWindow().innerWidth,
```
Comment (San4es) [inline js/core/utils/resize_callbacks.js]: Can we use window dimensions always?

File: testing/tests/DevExpress.core/utils.resize_callbacks.tests.js [hunk 2]
```diff
@@ -86,6 +87,26 @@ QUnit.test('Callback is not called if size is not changed', function(assert) {
...
 
+QUnit.test('Callback is called if window innerHeight is changed (T834502)', function(assert) {
+    let callCount = 0;
+    this.callbacks.add(function() {
+        callCount++;
+    });
+    const originalGetWindow = windowUtils.getWindow;
+
+    try {
+        windowUtils.getWindow = function() {
```
Comment (San4es) [inline testing/tests/DevExpress.core/utils.resize_callbacks.tests.js]: Use sinon.stub?

File: testing/tests/DevExpress.core/utils.resize_callbacks.tests.js [hunk 3]
```diff
@@ -86,6 +87,26 @@ QUnit.test('Callback is not called if size is not changed', function(assert) {
     assert.ok(!called, 'callback is not called');
 });
 
+QUnit.test('Callback is called if window innerHeight is changed (T834502)', function(assert) {
+    let callCount = 0;
+    this.callbacks.add(function() {
```
Comment (San4es) [inline testing/tests/DevExpress.core/utils.resize_callbacks.tests.js]: Use sinon.spy?

File: testing/tests/DevExpress.ui.widgets/popup.tests.js [hunk 4]
```diff
@@ -764,6 +765,34 @@ QUnit.test("popup height should support TreeView with Search if height = auto (T
...
+    }).dxPopup("instance");
+
+    const originalGetWindow = windowUtils.getWindow;
+    const $popup = instance.$content().parent();
+    const $popupContent = instance.$content();
+    const topToolbarHeight = $popup.find(toSelector(POPUP_TITLE_CLASS)).eq(0).innerHeight() || 0;
+    const bottomToolbarHeight = $popup.find(toSelector(POPUP_BOTTOM_CLASS)).eq(0).innerHeight() || 0;
+
+    try {
+        windowUtils.getWindow = function() {
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets/popup.tests.js]: Use sinon.stub?

File: testing/tests/DevExpress.ui.widgets/popup.tests.js [hunk 5]
```diff
@@ -764,6 +765,34 @@ QUnit.test("popup height should support TreeView with Search if height = auto (T
...
+            return { innerHeight: 100, innerWidth: 200 };
+        };
+
+        window.dispatchEvent(new Event('resize'));
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets/popup.tests.js]: Trigger resizeCallbacks?

File: testing/tests/DevExpress.core/utils.resize_callbacks.tests.js [hunk 6]
```diff
@@ -86,6 +87,26 @@ QUnit.test('Callback is not called if size is not changed', function(assert) {
...
+    try {
+        windowUtils.getWindow = function() {
+            return { innerHeight: 100, innerWidth: 100 };
+        };
+
+        this.triggerResize(false);
+
+        assert.strictEqual(callCount, 1, 'callback is called');
+    } finally {
+        windowUtils.getWindow = originalGetWindow;
```
Comment (San4es) [inline testing/tests/DevExpress.core/utils.resize_callbacks.tests.js]: Remove the test callback added [here](https://github.com/DevExpress/DevExtreme/pull/11070/files#diff-c2fcb0648b6ef84d628bc321b8d1eae7R92)?

File: testing/tests/DevExpress.core/utils.resize_callbacks.tests.js [hunk 7]
```diff
@@ -86,6 +86,21 @@ QUnit.test('Callback is not called if size is not changed', function(assert) {
...
+    this.callbacks.add(spy);
+
+    try {
+        sinon.stub(windowUtils, "getWindow").returns({ innerHeight: 100, innerWidth: 200 });
+
+        this.triggerResize(false);
+
+        assert.strictEqual(spy.callCount, 1, 'callback is called');
+    } finally {
+        windowUtils.getWindow.restore();
```
Comment (San4es) [inline testing/tests/DevExpress.core/utils.resize_callbacks.tests.js:100]: Remove added [here](https://github.com/DevExpress/DevExtreme/pull/11070/files#diff-c2fcb0648b6ef84d628bc321b8d1eae7R91) callback.

File: js/core/utils/resize_callbacks.js [hunk 8]
```diff
@@ -15,10 +15,10 @@ var resizeCallbacks = (function() {
     }
 
     var formatSize = function() {
-        var documentElement = domAdapter.getDocumentElement();
+        var window = windowUtils.getWindow();
         return {
-            width: documentElement.clientWidth,
-            height: documentElement.clientHeight
+            width: window.innerWidth,
```
Comment (San4es) [inline js/core/utils/resize_callbacks.js:20]: @nightskylark these changes look good to you?

Comment (nightskylark) [reply js/core/utils/resize_callbacks.js:20]: I recommend to check it on different jquery versions.

Comment (nightskylark) [reply js/core/utils/resize_callbacks.js:20]: And I'm not sure it's well covered by tests. so it would be great to ensure it works fine manually.

Comment (alexander-kotov-dx) [reply js/core/utils/resize_callbacks.js:20]: I have check most common cases with some jquery versions (3.4.1, 3.1.0, 2.2.1, 2.1.2) on android and ios devices (iphone 10), desktop chrome, ie, macOS safari and do not find any regressions.

**Review by San4es — APPROVED** [review]
LGTM

**Review by nightskylark — APPROVED** [review]

**Review by ksercs — APPROVED** [review]

**Pull request merged**